src/main/java/com/checkmarx/flow/service/JiraService.java

```diff
@@ -745,25 +744,37 @@ private Transition getTransitionByName(Iterable<Transition> transitions, String
         return null;
     }
 
-    private void loadCustomFields() {
+    private void loadCustomFields(String jiraProject, String issueType, boolean force) {
         log.debug("Loading all custom fields");
+        Map<String, String> fields = new HashMap<>();
+        if(!force && this.customFields.containsKey(jiraProject.concat(issueType))){
+            return;
+        }
         GetCreateIssueMetadataOptions options;
         options = new GetCreateIssueMetadataOptionsBuilder()
                 .withExpandedIssueTypesFields()
+                .withProjectKeys(jiraProject)
+                .withIssueTypeNames(issueType)
                 .build();
         Iterable<CimProject> metadata = this.issueClient.getCreateIssueMetadata(options).claim();
         CimProject cim = metadata.iterator().next();
         cim.getIssueTypes().forEach( issueTypes -> {
             issueTypes.getFields().forEach((id, value) -> {
                 String name = value.getName();
-                this.customFields.put(name, id);
+                fields.put(name, id);
             });
         });
+        this.customFields.put(jiraProject.concat(issueType), fields);
     }
 
-    private String getCustomFieldByName(String fieldName) {
+    private String getCustomFieldByName(String jiraProject, String issueType, String fieldName) {
         log.debug("Getting custom field {}", fieldName);
-        return this.customFields.get(fieldName);
+        //TODO logic for forcing a refresh of custom fields
```

Comment:
adding this comment to make sure you are aware of this TODO and intentionally didn't implement it

Reply:
This fix had to get out quickly for a customer.  This is left intentionally.  The intention is to come up with an approach to regularly refresh the cache (TBD) on a periodic basis in case the custom fields change.  It is low priority as a restart solves this, but long term should be a better solution.